mne_neuron/dipole.py

```diff
@@ -14,22 +16,29 @@ def _hammfilt(x, winsz):
     return convolve(x, win, 'same')
 
 
-def simulate_dipole(net):
-    """Simulate a dipole given the experiment parameters.
+def _clone_and_simulate(params, trial_idx):
+    from .network import Network
 
-    Parameters
-    ----------
-    net : Network object
-        The Network object specifying how cells are
-        connected.
+    if trial_idx != 0:
+        params['prng_*'] = trial_idx
+
+    net = Network(params, n_jobs=1)
+
+    from neuron import h
+    net._create_all_src()
+    net.state_init()
+    net._parnet_connect()
+    # set to record spikes
+    net.spiketimes = h.Vector()
+    net.spikegids = h.Vector()
+    net._record_spikes()
+
```

Comment:
@jasmainak I'm preparing a PR to add ParallelContext (n_cores > 1, but not compatible with n_jobs > 1). A function similar to _clone_and_simulate will be called, and the net.* functions need to be part of it. Can you explain why these functions need to separate from Network(params, n_cores)? All of _clone_and_simulate gets pickled, right? Perhaps these net.* calls can be part of an appropriately named method of Network.